pkg/kube/controllers/extendedjob/job_create_secret.go

```diff
@@ -0,0 +1,286 @@
+
+	secretName := namePrefix + outputContainer.Name
+
+	// Fetch json from file
+	filePath := fmt.Sprintf("%s%s%s", "/mnt/quarks/", outputContainer.Name, "/output.json")
+	file, _ := ioutil.ReadFile(filePath)
+	var data map[string]string
+	err := json.Unmarshal([]byte(file), &data)
+	if err != nil {
+		return errors.Wrapf(err, "failed to convert output file %s into json for creating secret %s in pod %s",
+			filePath, secretName, podName)
+	}
+
+	// Create secret for the outputfile to persist
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+	}
+	secretLabels := exjob.Spec.Output.SecretLabels
+	if secretLabels == nil {
+		secretLabels = map[string]string{}
+	}
+	secretLabels[ejv1.LabelPersistentSecretContainer] = outputContainer.Name
+	if ig, ok := podutil.LookupEnv(outputContainer.Env, converter.EnvInstanceGroupName); ok {
+		secretLabels[ejv1.LabelInstanceGroup] = ig
+	}
+
+	if exjob.Spec.Output.Versioned {
+		// Use secretName as versioned secret name prefix: <secretName>-v<version>
+		err = createVersionSecret(clientSet, namespace, exjob.GetName(), exjob.GetUID(), secretName, data, secretLabels, "created by extendedjob")
+		if err != nil {
+			return errors.Wrapf(err, "could not persist ejob's %s output to a secret", exjob.GetName())
+		}
+	} else {
+		secret.StringData = data
+		secret.Labels = secretLabels
+
+		_, err = clientSet.CoreV1().Secrets(namespace).Create(secret)
+
+		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				// If it exists update it
+				_, err = clientSet.CoreV1().Secrets(namespace).Update(secret)
+				if err != nil {
+					return errors.Wrapf(err, "Failed to update secret %s for container %s in pod %s.", secretName, outputContainer.Name, podName)
+				}
+			} else {
+				return errors.Wrapf(err, "Failed to create secret %s for container %s in pod %s.", secretName, outputContainer.Name, podName)
+			}
+		}
+
+	}
+	return nil
+}
+
+func createVersionSecret(clientSet *kubernetes.Clientset, namespace string, ownerName string, ownerID types.UID, secretName string, secretData map[string]string, labels map[string]string, sourceDescription string) error {
+	currentVersion, err := getGreatestVersion(clientSet, namespace, secretName)
+	if err != nil {
+		return err
+	}
+
+	version := currentVersion + 1
+	labels[versionedsecretstore.LabelVersion] = strconv.Itoa(version)
+	labels[versionedsecretstore.LabelSecretKind] = versionedsecretstore.VersionSecretKind
+
+	generatedSecretName, err := versionedsecretstore.GenerateSecretName(secretName, version)
+	if err != nil {
+		return err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generatedSecretName,
+			Namespace: namespace,
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "fissile.cloudfoundry.org/v1alpha1",
```

Comment:
please use a constant - we already define this

Reply:
Done.